packages/lesswrong/server/datadog/datadogMiddleware.ts

```diff
@@ -0,0 +1,20 @@
+import tracer from './tracer'
+
+/**
+ * - Attach user info to the root span
+ * - Allow the headers required for Real User Monitoring
+ */
+export const datadogMiddleware = (req, res, next) => {
+  const span = tracer.scope().active()
+  if (span !== null) {
+    // @ts-ignore - there is currently no public API for getting the root span, this is the accepted way (see https://github.com/DataDog/dd-trace-js/issues/725#issuecomment-805277510)
+    const rootSpan = span.context()._trace.started[0]
+    const user = req.user
+    if (user) {
+      rootSpan.setTag('usr', {id: user._id, name: user.displayName, email: user.email, slug: user.slug})
```

Comment:
`usr: {id, name, email}` is the preferred format for user info in datadog 🤷‍♂️

Reply:
Does this need to match the user data in `configureDatadogRum()`? The `email` values could be different. Also, if they do need to be kept in sync, could you add comments in the code explaining that?